codec-http/src/main/java/io/netty/handler/codec/http/QueryStringDecoder.java

```diff
@@ -224,6 +280,10 @@ private void decodeParams(String s) {
                     name = decodeComponent(s.substring(pos, i), charset);
                 }
                 pos = i + 1;
+            } else if (withCommaDecoding && c == ',' && name != null) {
```

Comment:
The more I think about it, the more I think this change is a bad idea.
For example, one could use an URI such as `?foo=a,b;c=d` to mean `foo=[["a", "b"], ["c", "d"]]`. If we were to support such valid usage, we would need `parameters()` to return a `Map<String, List<List<String>>>`, which is not possible without breaking the API.

All in all, sub-delimiters usage in query is not specified and application specific. I think it would be better to let such parsing up to the user.